src/main/scala/is/hail/rvd/OrderedRVDPartitioner.scala

```diff
@@ -105,6 +109,33 @@ class OrderedRVDPartitioner(
     new OrderedRVDPartitioner(kType, newBounds, allowedOverlap)
   }
 
+  def intersect(other: OrderedRVDPartitioner): OrderedRVDPartitioner = {
+    require(kType isIsomorphicTo other.kType)
+
+    val kord = kType.ordering
+    val eord = kord.intervalEndpointOrdering.toOrdering.asInstanceOf[Ordering[IntervalEndpoint]]
+
+    val left = this.rangeBounds.iterator.buffered
+    val right = other.rangeBounds.iterator.buffered
+    val ab = new ArrayBuilder[Interval]()
+
+    while (left.hasNext && right.hasNext) {
```

Comment:
if one side is empty, this returns an empty partitioner. That's correct behavior, right?

Reply:
For intersection, yeah. For example, if you do an inner join with an RVD with no partitions, then this will give a joined RVD with no partitions.